scripts/agent_transport_probe.py

```
@@ -0,0 +1,88 @@
+
+
+def checksum(data):
+    return binascii.crc_hqx(data, 0xFFFF)
+
+
+def read_exact(stream, size):
+    data = bytearray()
+    while len(data) < size:
+        chunk = stream.recv(size - len(data))
+        if not chunk:
+            raise ProtocolError("partial frame")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)